### ISSUE TITLE
Support jsonschema3

### ISSUE THREAD

**Issue by jgarver** [zalando/connexion#983]
It looks like we can support jsonschema3 easily now.

setup.py forces 'jsonschema>=2.5.1,<3.0.0'.  It looks like this came in with openapi-spec-validator, which required jsonschema<3.0.0.  In the meantime, openapi-spec-validator has added support for jsonschema3 starting with 0.2.7.

As a quick test, I bumped jsonschema up to 3.0.1 and connexion's unittests still pass.

**Comment by chrish42** [comment]
That would be great, given that some other well-known projects (e.g. jupyterlab) have started requiring jsonschema >= 3.0.1, making it hard to use them together with a connextion-using library / app.

**Comment by danie1k** [comment]
I've made an attempt to update `jsonschema` package to newest version and use it in my pre-production-ready project from fork.  
I've faced a few problems which needs to be solved IMO, before connexion's requirement could be updated.

1. **Should such update also came with support for [JSON Schema Draft-07](https://json-schema.org/specification.html)?** - TBH, I lack the competence to definite answer to answer this question.
    _@hjacobs, @dtkav you've participated in my PR (#996), are you able to help here?_
2. Next actions depends on decision between:
    1. staying with [officially abandoned draft-05](https://json-schema.org/draft-06/json-schema-release-notes.html#q-what-happened-to-draft-05) 
      or
    2. waitng for third-parties to be updated, to support draft-07.

----

I **haven't tried** first option - just updating `jsonschema` package to it's newest version. Maybe it will go smoothly and will be enough.  
But it will result in situation, where package `jsonschema` will support draft-07, while `openapi-spec-validator` & `connextion` won't.

On the other hand, updating to draft-07 will be very hard & time consuming, because connexion relies on lots of third-party dependencies.  
I've already tried some more complex updates:
* zalando/connexion#996
* p1c2u/openapi-spec-validator#78

And again, seems that my knowledge about OpenApi & JSON Schema is not good enoug - you definetely should check following comment: https://github.com/p1c2u/openapi-spec-validator/pull/78#issuecomment-513784804
